### ISSUE TITLE
`s3.get_object` `IfModifiedSince` default should `...`

### ISSUE THREAD

**Issue by sbdchd** [vemel/mypy_boto3_builder#110]
**Describe the bug**

I think the default for this param, and maybe most params should be `...` instead of `None` to avoid type checkers like `pyright` from changing the type to include a union of `None`.

Mypy doesn't add `None` to the type so using `None` is fine in that case. Although I think there is a setting to change this behavior.

**To Reproduce**
1. Install 'boto3-stubs[...]'
2.  Run mypy on the following code sample

```python
import boto3

s3 = boto3.client("s3")
s3.get_object(IfModifiedSince=None)
```
3. Actual mypy output

```
error: Argument "IfModifiedSince" to "get_object" of "S3Client" has incompatible type "None"; expected "datetime"
```

**Expected mypy-output**

```
N/A
```

**Additional context**
boto3-stubs: 1.18.49

**Edit:**
Essentially instead of:

```python
def get_object(
    self,
    *,
    Bucket: str,
    Key: str,
    IfMatch: str = None,
    IfModifiedSince: Union[datetime, str] = None,
    IfNoneMatch: str = None,
    IfUnmodifiedSince: Union[datetime, str] = None,
    Range: str = None,
    ResponseCacheControl: str = None,
    ResponseContentDisposition: str = None,
    ResponseContentEncoding: str = None,
    ResponseContentLanguage: str = None,
    ResponseContentType: str = None,
    ResponseExpires: Union[datetime, str] = None,
    VersionId: str = None,
    SSECustomerAlgorithm: str = None,
    SSECustomerKey: str = None,
    SSECustomerKeyMD5: str = None,
    RequestPayer: Literal["requester"] = None,
    PartNumber: int = None,
    ExpectedBucketOwner: str = None
) -> GetObjectOutputTypeDef:
  ...
```

I think we should have:

```python
def get_object(
    self,
    *,
    Bucket: str,
    Key: str,
    IfMatch: str = ...,
    IfModifiedSince: Union[datetime, str] = ...,
    IfNoneMatch: str = ...,
    IfUnmodifiedSince: Union[datetime, str] = ...,
    Range: str = ...,
    ResponseCacheControl: str = ...,
    ResponseContentDisposition: str = ...,
    ResponseContentEncoding: str = ...,
    ResponseContentLanguage: str = ...,
    ResponseContentType: str = ...,
    ResponseExpires: Union[datetime, str] = ...,
    VersionId: str = ...,
    SSECustomerAlgorithm: str = ...,
    SSECustomerKey: str = ...,
    SSECustomerKeyMD5: str = ...,
    RequestPayer: Literal["requester"] = ...,
    PartNumber: int = ...,
    ExpectedBucketOwner: str = ...
) -> GetObjectOutputTypeDef:
  ...
```

**Comment by vemel** [comment]
Thank you for the report!

The problem is, that builder generates both `*.pyi` files where `...` is acceptable and `*.py` files to allow users to import types. 

A proper fix could be to render argument default values as `None` to `py` and as `...` to `pyi`.

I will try to address this in the next version.

**Comment by vemel** [comment]
There is a reason why argument types are not wrapped in `Optional`. In many cases it is not safe to provide `None` value to boto3 method calls. Unfortunately, there is no way to figure out from services shapes if `None` can be provided or not, so it is safer to not pass optional arguments if you want default values.

**Comment by vemel** [comment]
Addressed in https://github.com/vemel/mypy_boto3_builder/releases/tag/5.5.0 , so it should be included in `boto3-stubs 1.18.51` release.

Please let me know if it works as it should.
